### PR TITLE
[Core: Utility]  Add type declarations to function signatures

### PR DESCRIPTION
**Pull request by johnsaigle**
See #3878 for a more detailed description.

This PR also fixes problematic calling classes in the statistics module.

### CODE REVIEW

File: php/libraries/Utility.class.inc [hunk 1]
```diff
@@ -38,17 +38,18 @@ class Utility
...
      * @param string $testdate date of testing (or second date), formatted YYYY-MM-DD
      *
-     * @return mixed an array with keys 'year', 'mon', 'day' containing the
-     *               computed age, or a PEAR::Error object
+     * @return array containing keys 'year', 'mon', 'day' containing the
+     *               computed age
+     * @throws LorisException
      */
-    static function calculateAge($dob, $testdate)
+    static function calculateAge($dob, $testdate): array
```
Comment (PapillonMcGill) [inline php/libraries/Utility.class.inc]: ...(string $dbo, string $testdate)...

File: php/libraries/Utility.class.inc [hunk 2]
```diff
@@ -117,9 +118,9 @@ class Utility
      *
      * @return array an associative array("center ID" => "site name")
      */
-    static function getSiteList($study_site = true)
+    static function getSiteList(bool $study_site = true): array
```
Comment (PapillonMcGill) [inline php/libraries/Utility.class.inc]: bool|null in the doc block

File: php/libraries/Utility.class.inc [hunk 3]
```diff
@@ -150,9 +151,11 @@ class Utility
      *         should be interpreted as an associative array since the keys
      *         refer to the centerID, not the array index.
      */
-    static function getAssociativeSiteList($study_site = true, $DCC = true)
-    {
-        $DB =& Database::singleton();
+    static function getAssociativeSiteList(
+        bool $study_site = true,
+        bool $DCC = true
+    ): array {
```
Comment (PapillonMcGill) [inline php/libraries/Utility.class.inc]: same here

File: php/libraries/Utility.class.inc [hunk 4]
```diff
@@ -425,14 +430,14 @@ class Utility
...
      *
-     * @param integer $age   age in days
-     * @param string  $stage study stage (screening or visit)
+     * @param integer     $age   age in days
+     * @param string|null $stage study stage (screening or visit)
      *
      * @return array list of instruments
      */
-    static function lookupBattery($age, $stage=null)
+    static function lookupBattery(int $age, string $stage = null): array
```
Comment (PapillonMcGill) [inline php/libraries/Utility.class.inc]: **?** string $stage

File: php/libraries/Utility.class.inc (hunk 2 above)
Comment (johnsaigle) [reply php/libraries/Utility.class.inc]: As discussed in the other PR, this param isn't actually nullable.

File: php/libraries/Utility.class.inc (hunk 3 above)
Comment (johnsaigle) [reply php/libraries/Utility.class.inc]: same here. :)

File: php/libraries/Utility.class.inc [hunk 5]
```diff
@@ -216,12 +219,12 @@ class Utility
...
      * Returns a list of study Subprojects
      *
-     * @param integer $projectID The project for which you would like
+     * @param integer|null $projectID The project for which you would like
      *                           to get the subprojects
      *
      * @return array       an associative array("SubprojectID" => "Subproject title")
      */
-    static function getSubprojectList($projectID = null)
+    static function getSubprojectList(int $projectID = null): array
```
Comment (PapillonMcGill) [inline php/libraries/Utility.class.inc]: I know Travis is happy but should't it be `?int $projectID = null` (same for line 257)?

**Pull request merged**